It builds, but does not work.

```
building '/nix/store/q781h2mz5pwgvcmg4wwhkly43llc85zp-python3-3.9.6-env.drv'...
created 283 symlinks in user environment
Python 3.9.6 (default, Jun 28 2021, 08:57:49) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import wx
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/187bwr9g29405apfm7l6hr3jcsdbcfkm-python3-3.9.6-env/lib/python3.9/site-packages/wx/__init__.py", line 17, in <module>
    from wx.core import *
  File "/nix/store/187bwr9g29405apfm7l6hr3jcsdbcfkm-python3-3.9.6-env/lib/python3.9/site-packages/wx/core.py", line 12, in <module>
    from ._core import *
ImportError: libwx_gtk3u_core-3.1.so.5: cannot open shared object file: No such file or directory
```

--- a/pkgs/development/python-modules/wxPython/4.1.nix
+++ b/pkgs/development/python-modules/wxPython/4.1.nix
@@ -13,29 +13,59 @@
 , wxGTK
 , pillow
 , numpy
+, libXinerama
+, libSM
+, libXxf86vm
+, libXtst
+, libGLU
+, libGL
+, xorgproto
+, gst_all_1
+, libglvnd
+, mesa
+, webkitgtk
+, autoPatchelfHook
 }:
 let
   dynamic-linker = stdenv.cc.bintools.dynamicLinker;
 in
 buildPythonPackage rec {
   pname = "wxPython";
-  version = "4.1.0";
+  version = "4.1.1";
   disabled = isPy27;
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "12x4ci5q7qni4rkfiq6lnpn1fk8b0sfc6dck5wyxkj2sfp5pa91f";
+    sha256 = "0a1mdhdkda64lnwm1dg0dlrf9rs4gkal3lra6hpqbwn718cf7r80";
   };
 
   # https://github.com/NixOS/nixpkgs/issues/75759
   # https://github.com/wxWidgets/Phoenix/issues/1316
   doCheck = false;
 
-  nativeBuildInputs = [ which doxygen wxGTK pkg-config ];
+  nativeBuildInputs = [
+    which
+    doxygen
+    wxGTK.gtk
+    pkg-config
+    autoPatchelfHook
+  ];
 
   buildInputs = [
     wxGTK.gtk
     ncurses
+    libXinerama
+    libSM
+    libXxf86vm
+    libXtst
+    xorgproto
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    libGLU
+    libGL
+    libglvnd
+    mesa
+    webkitgtk
   ];
 
   propagatedBuildInputs = [ pillow numpy ];
@@ -44,30 +74,22 @@ buildPythonPackage rec {
 
   preConfigure = lib.optionalString (!stdenv.isDarwin) ''
     substituteInPlace wx/lib/wxcairo/wx_pycairo.py \
-      --replace 'cairoLib = None' 'cairoLib = ctypes.CDLL("${cairo}/lib/libcairo.so")'
-    substituteInPlace wx/lib/wxcairo/wx_pycairo.py \
       --replace '_dlls = dict()' '_dlls = {k: ctypes.CDLL(v) for k, v in [
         ("gdk",        "${wxGTK.gtk}/lib/libgtk-x11-3.0.so"),
         ("pangocairo", "${pango.out}/lib/libpangocairo-1.0.so"),
+        ("cairoLib = None", "cairoLib = ctypes.CDLL('${cairo}/lib/libcairo.so')"),
         ("appsvc",     None)
       ]}'
-
-    # https://github.com/wxWidgets/Phoenix/pull/1584
-    # It'll be not needed in next version because https://github.com/wxWidgets/Phoenix/pull/1699
-    substituteInPlace build.py --replace "os.environ['PYTHONPATH'] = phoenixDir()" \
-      "os.environ['PYTHONPATH'] = os.environ['PYTHONPATH'] + os.pathsep + phoenixDir()"
   '';
 
   buildPhase = ''
-    ${python.interpreter} build.py -v --use_syswx dox etg --nodoc build_py
+    ${python.interpreter} build.py -v build_wx dox etg --nodoc sip build_py
   '';
 
   installPhase = ''
     ${python.interpreter} setup.py install --skip-build --prefix=$out
     wrapPythonPrograms
   '';
-
-  passthru = { inherit wxGTK; };
 
   meta = with lib; {
     description = "Cross platform GUI toolkit for Python, Phoenix version";

